crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -30,6 +33,126 @@ async fn local_node_cow_amm() {
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(300_000), to_wei(100))
+        .await;
+
+    // set up cow_amm
+    let oracle = contracts::CowAmmUniswapV2PriceOracle::builder(&web3)
+        .deploy()
+        .await
+        .unwrap();
+
+    let cow_amm_factory = contracts::CowAmmConstantProductFactory::builder(
+        &web3,
+        onchain.contracts().gp_settlement.address(),
+    )
+    .deploy()
+    .await
+    .unwrap();
+
+    let block_stream = ethrpc::current_block::current_block_stream(
+        NODE_HOST.parse().unwrap(),
+        Duration::from_millis(1_000),
+    )
+    .await
+    .unwrap();
+
+    let block_retriever: Arc<dyn BlockRetrieving> = Arc::new(web3.clone());
+    let cow_amm_indexer = Registry::build(
+        block_retriever,
+        CowAmmSafeBasedContract::new(cow_amm_factory.clone()),
+        block_stream,
+        cow_amm_factory.deployment_information(),
+    )
+    .await;
+
+    // Fund cow amm owner with 2_000 dai and allow factory take them
+    dai.mint(cow_amm_owner.address(), to_wei(2_000)).await;
+    tx!(
+        cow_amm_owner.account(),
+        dai.approve(cow_amm_factory.address(), to_wei(2_000))
+    );
+    // Fund cow amm owner with 1 WETH and allow factory take them
+    tx_value!(
+        cow_amm_owner.account(),
+        to_wei(1),
+        onchain.contracts().weth.deposit()
+    );
+    tx!(
+        cow_amm_owner.account(),
+        onchain
+            .contracts()
+            .weth
+            .approve(cow_amm_factory.address(), to_wei(1))
+    );
+
+    let pair = onchain
+        .contracts()
+        .uniswap_v2_factory
+        .get_pair(onchain.contracts().weth.address(), dai.address())
+        .call()
+        .await
+        .expect("failed to get Uniswap V2 pair");
+
+    let cow_amm = cow_amm_factory
+        .amm_deterministic_address(
+            cow_amm_owner.address(),
+            dai.address(),
+            onchain.contracts().weth.address(),
+        )
+        .call()
+        .await
+        .unwrap();
+
+    let oracle_data: Vec<_> = std::iter::repeat(0u8)
+        .take(12) // pad with 12 zeros in the front to end up with 32 bytes
+        .chain(pair.as_bytes().to_vec())
+        .collect();
+    const APP_DATA: [u8; 32] = [12u8; 32];
+
+    cow_amm_factory
+        .create(
+            dai.address(),
+            to_wei(2_000),
+            onchain.contracts().weth.address(),
+            to_wei(1),
+            0.into(), // min traded token
+            oracle.address(),
+            ethcontract::Bytes(oracle_data.clone()),
+            ethcontract::Bytes(APP_DATA),
+        )
+        .from(cow_amm_owner.account().clone())
+        .send()
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async {
+        let cows = cow_amm_indexer.cow_amms().await;
+        !cows.is_empty()
+    })
+    .await
+    .unwrap();
+
+    wait_for_condition(TIMEOUT, || async {
+        let cows = cow_amm_indexer.cow_amms().await;
```

Comment:
Testing that the component picks up these addresses is less code coverage than the test you had before (asserting that the cow amm addresses actually make it into the auction sent to the driver).

Reply:
@MartinquaXD this test is intended to just test the indexer/registry. I can return back the previous test too if you want, but I see if superfluous since it will be tested more throroughly in the upcoming PRs.

Reply:
I meant testing that the automatically indexed cow amm address makes it into the auction verifies that the component is able to find it but also that the addresses get added to the auction correctly.
Overall e2e tests should try to test at the highest abstraction possible. That way we generally:
* test more code paths
* test what we actually care about
* require less refactoring when implementation details or APIs change

This guiding principle has served us pretty well for e2e tests.

Reply:
I actually removed the `fee-exempt` addresses from the autopilot setgging, which automatically test what you mentioned, otherwise the test would fail

Reply:
Not sure I understand. That test passed at some point, no?
Why does it suddenly cause issues?

Reply:
What was the resolution here? IMO there is no need to create another e2e test for this little detail if we could also assert this in the existing cow amm e2e test.